app/src/main/java/com/codelab/android/datastore/ui/TasksActivity.kt

```diff
@@ -62,12 +62,13 @@ class TasksActivity : AppCompatActivity() {
 
         setupRecyclerView()
         setupFilterListeners(viewModel)
-        setupSort()
 
         viewModel.tasksUiModel.observe(this) { tasksUiModel ->
             adapter.submitList(tasksUiModel.tasks)
             updateSort(tasksUiModel.sortOrder)
             binding.showCompletedSwitch.isChecked = tasksUiModel.showCompleted
+            // Setup change listeners after the initial emission to avoid triggering a re-emission
+            setupOnCheckedChangeListeners()
```

Comment:
Move back to onCreate and address the real problem: 

When you modify `binding.sortDeadline.isChecked` in `updateSort`, the listener set with `setOnCheckedChangeListener` is called because it doesn't distinguish between a user set or a programmatic set which could be the reason for the infinite loop that we were seeing. However, if `updateSort` is setting a new value that is contradicted by `viewModel.enableSortByDeadline(checked)`, there must be a bigger problem. Can you look into this?